src/optimal_retention.rs

```diff
@@ -233,16 +223,26 @@ pub fn simulate(
 
         let is_learn = card.last_date == f32::NEG_INFINITY;
 
+        let last_date_index = card.last_date as usize;
+
         // Guards
         if card.due >= learn_span as f32 {
+            if !is_learn {
+                let delta_t = learn_span - last_date_index;
```

Comment:
The cards in one of my very out of date decks breaks it.
Add
```rs
Card {
    difficulty: 6.2065673,
    stability: 479.77164,
    last_date: 299.0,
    due: 956.0,
}
```
to the existing cards test and it panics:
```
[src/optimal_retention.rs:231:17] &learn_span = 30
[src/optimal_retention.rs:231:17] &last_date_index = 299
thread 'optimal_retention::tests::simulate_with_existing_cards' panicked at src/optimal_retention.rs:231:31:
attempt to subtract with overflow
```

Reply:
? Why the last_date is larger than zero? I think this input is invalid.

Reply:
```rs
[rslib/src/scheduler/fsrs/simulator.rs:89:29] &due = 1792
[rslib/src/scheduler/fsrs/simulator.rs:89:29] &days_elapsed = 836
[rslib/src/scheduler/fsrs/simulator.rs:89:29] &relative_due = 956
[rslib/src/scheduler/fsrs/simulator.rs:89:29] &card = Card {
    id: CardId(
        1702031515935,
    ),
    note_id: NoteId(
        1702031515933,
    ),
    deck_id: DeckId(
        1675956055668,
    ),
    template_idx: 1,
    mtime: TimestampSecs(
        1731253620,
    ),
    usn: Usn(
        -1,
    ),
    ctype: Review,
    queue: Review,
    due: 1792,
    interval: 1,
    ease_factor: 1300,
    reps: 1376,
    lapses: 71,
    remaining_steps: 0,
    original_due: 0,
    original_deck_id: DeckId(
        0,
    ),
    flags: 0,
    original_position: Some(
        8709,
    ),
    memory_state: Some(
        FsrsMemoryState {
            stability: 0.13,
            difficulty: 10.0,
        },
    ),
    desired_retention: Some(
        0.87,
    ),
    custom_data: "{\"v\": \"reschedule\"}",
}
[rslib/src/scheduler/fsrs/simulator.rs:89:29] FSRSCard = Card {
    cid: 1702031515935,
    difficulty: 10.0,
    stability: 0.13,
    last_date: 955.0,
    due: 956.0,
}
```
![image](https://github.com/user-attachments/assets/d610b644-e3b8-4c59-a38a-0581c1bb6e6c)

I thought filtered reviews were supposed to be filtered out?
Moving the card to a different deck fixes it (?!?).

This is an obscure, pretty much unrelated Anki problem, you can merge if you want.